src/proposals/proposals.controller.ts

```diff
@@ -215,49 +344,86 @@ export class ProposalsController {
     );
   }
 
-  // DELETE /proposals/:id/attachments/:fk
+  // DELETE /proposals/:pid/attachments/:aid
   @UseGuards(PoliciesGuard)
   @CheckPolicies((ability: AppAbility) =>
     ability.can(Action.Delete, Attachment),
   )
-  @Delete("/:id/attachments/:fk")
+  @Delete("/:pid/attachments/:aid")
+  @ApiParam({
+    name: "pid",
+    description: "Id of the proposal for which we would like to delete the attachment specified",
+    type: String,
+  })
+  @ApiParam({
+    name: "aid",
+    description: "Id of the attachment of this proposal that we would like to delete",
+    type: String,
+  })
+  @ApiResponse({
+    status: 200,
+    description: "Remove the attachment with id specified associated with the proposal with the pid specified"
+  })
   async findOneAttachmentAndRemove(
-    @Param("id") proposalId: string,
-    @Param("fk") attachmentId: string,
+    @Param("pid") proposalId: string,
+    @Param("aid") attachmentId: string,
   ): Promise<unknown> {
     return this.attachmentsService.findOneAndRemove({
       _id: attachmentId,
       proposalId: proposalId,
     });
   }
 
-  // POST /proposals/:id/datasets
-  @UseGuards(PoliciesGuard)
+  // POST /proposals/:pid/datasets
+/*   @UseGuards(PoliciesGuard)
```

Comment:
I see you have commented this part and if it is not needed anymore maybe removing it is better.

Reply:
Done. We want to provide only the functionality to return the linked datasets, but not to create/update datasets from proposals